File: src/main/java/com/rss/config/properties/DatasourceProperties.java
package com.rss.config.properties;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties(prefix = "db")
public class DatasourceProperties {
    private String uri;
    private String username;
    private String password;
    private String schema;

    public String getUri() {
        return uri;
    }

    public void setUri(String uri) {
        this.uri = uri;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getSchema() {
        return schema;
    }

    public void setSchema(String schema) {
        this.schema = schema;
    }
}

File: src/main/java/com/rss/config/properties/DatasourceProperties.kt
package com.rss.config.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "db")
open class DatasourceProperties {
    var uri: String? = null
    var username: String? = null
    var password: String? = null
    var schema: String? = null
}